src/projectscene/qml/Audacity/ProjectScene/common/InsertSilence.qml

```diff
@@ -0,0 +1,100 @@
+import QtQuick
+import QtQuick.Controls
+import QtQuick.Layouts
+import Muse.UiComponents
+import Audacity.Effects
+import Audacity.Playback
+import Audacity.ProjectScene
+
+StyledDialogView {
+    id: root
+
+    title: qsTrc("effects/silence", "Silence")
+
+    property double duration: 0
+    property alias durationFormat: timecode.currentFormatStr
+
+    contentWidth: 320
+    contentHeight: 150
+
+    function preview() {
+        silence.preview()
+    }
+
+    InsertSilenceModel {
+        id: silence
+    }
+
+    Component.onCompleted: {
+        silence.init()
+        silence.durationFormat = root.durationFormat
+        if(silence.duration == 0) {
+            silence.duration = root.duration
```

Comment:
could you please explain?

Reply:
If there is no selection, set duration to last used, provided by ProjectSceneActionsController

Reply:
Loading configuration can be moved to the model, what do you think?